# 14 · GSE232381 · bulk_RNA_seq · WGCNA modules

Reads `wgcna_input.rds`. Writes `modules.rds` and the end product `module_genes.csv` in
`data/run_artifacts/GSE232381/`.

Settings as for GSE65391 (notebook 05): signed network and TOM, minModuleSize 30, mergeCutHeight 0.25.
There are more genes than fit one block, so `maxBlockSize = 5000`: WGCNA pre-clusters the genes into
blocks and builds modules within each block.

## Load WGCNA and the input from notebook 13

In [1]:
if (!requireNamespace("WGCNA", quietly = TRUE)) install.packages("WGCNA", repos = "https://cloud.r-project.org")
suppressMessages(library(WGCNA))
source("../src/paths.R")
w <- readRDS(art("GSE232381", "wgcna_input.rds"))
c(samples = nrow(w$datExpr), genes = ncol(w$datExpr), power = w$power)

samples   genes   power 
     25   13578      16

**Explanation**
- As in notebook 05, with the input of notebook 13.

## Find the modules

In [2]:
set.seed(SEED)
net <- blockwiseModules(w$datExpr, power = w$power, networkType = "signed", TOMType = "signed",
                        minModuleSize = 30, mergeCutHeight = 0.25, maxBlockSize = 5000,
                        numericLabels = FALSE, pamRespectsDendro = FALSE, verbose = 0)
sizes <- sort(table(net$colors), decreasing = TRUE)
c(modules = sum(names(sizes) != "grey"), genes_in_modules = sum(sizes[names(sizes) != "grey"]), grey = unname(sizes["grey"]))
sizes

modules genes_in_modules             grey 
              24            13273              305


    turquoise          blue         brown        yellow         green 
         3665          2501          2315          1346           446 
          red         black          pink          grey       magenta 
          427           375           331           305           261 
       purple   greenyellow           tan        salmon          cyan 
          194           193           178           174           138 
 midnightblue     lightcyan        grey60    lightgreen   lightyellow 
          135           132           101            59            59 
    royalblue       darkred     darkgreen darkturquoise      darkgrey 
           56            55            47            47            38 

**Explanation**
- `blockwiseModules()` as in notebook 05, with one difference: `maxBlockSize = 5000`. With more genes than
  that, WGCNA first splits the genes into blocks of at most 5,000 related genes and finds modules within each
  block. This limits the memory needed.
- The counts of modules, genes in modules and grey genes as in notebook 05.

**Result.** 24 modules holding 13,273 of 13,578 genes; 305 are grey.

## Module membership and hub genes

In [3]:
ME  <- orderMEs(net$MEs)
kME <- cor(w$datExpr, ME)
modules <- setdiff(names(sizes), "grey")
hubs <- lapply(setNames(modules, modules), function(mo) {
  g <- names(net$colors)[net$colors == mo]
  names(sort(kME[g, paste0("ME", mo)], decreasing = TRUE))
})
t(sapply(hubs, function(h) paste(head(h, 8), collapse = ", ")))

turquoise,blue,brown,yellow,green,red,black,pink,magenta,purple,⋯,midnightblue,lightcyan,grey60,lightgreen,lightyellow,royalblue,darkred,darkgreen,darkturquoise,darkgrey
"PTPN6, STXBP2, ALDOA, TALDO1, CHMP2A, OS9, CFP, EPN1","RASA2, FBXL3, WDR44, MTM1, GOLM2, CEP170, LTN1, ZBTB11","CEP290, HLTF, ZNF675, MSH2, ZNF567, USP37, NOC3L, ZWILCH","HAUS7, MRPL38, LONP1, PES1, MRPL4, ECHS1, DDX27, SCAMP3","C5AR2, RALGAPA2, SVIL, NPEPPS, CR1, MGAM, KLHL21, PFKFB2","CA11, NLE1, BCAS4, ALKBH2, SLC25A23, SLC27A5, LOC728743, MAGED1","PLA2G6, ITIH4, SH2B1, SLX1A-SULT1A3, SLX1B-SULT1A4, PTOV1-AS2, CAPN10, CAPS","NIFK, TOMM70, SHLD2, LYRM2, CEBPZ, GGCT, MPHOSPH10, TAF9B","CELF6, DNHD1, AGAP6, SH3BP5-AS1, STK36, SETD4, NPHP4, WDR19","ORC1, GMNN, ZWINT, RRM2, CDC6, MCM4, CDC45, SKA3",⋯,"GUCY1B1, LTBP1, ELOVL7, PLA2G12A, ASAP2, MFAP3L, ARHGAP21, PDE5A","MAFF, ZCCHC8, SRRM1, TNFAIP3, GZF1, NR4A2, KLF10, JUN","JMJD6, CFAP45, LINC00528, PARP6, SERTAD3, ZFAND2A, KMT5A, PPP1R15A","IFI44L, RSAD2, IFIT1, HERC5, HERC6, OAS3, IFIT3, OAS2","TM9SF3, ZYG11B, TBL1XR1, KIF5B, WAPL, RHOQ, CNOT6L, SSR1","CD28, AMIGO1, P2RY10, PAXIP1-AS2, RASGRF2, KIF21A, TRIB2, GDF11","SAP30L, RNF20, ZC3H4, STX6, NEK9, USP48, KMT5B, HIF1AN","GALE, DPM3, PFDN2, PMM1, EMC6, INO80C, PEX39, RP9P","CEACAM8, DEFA4, MPO, BPI, CTSG, LTF, OLFM4, CEACAM6","RRP1B, NIPAL3, DBT, MGA, ZFP90, UBIAD1, PPARA, RABEP1"


**Explanation**
- kME and the hub genes of each module as in notebook 05.

## Which modules hold the interferon genes?

In [4]:
ifn6 <- read_gene_set("ifn-type1-6.txt")
data.frame(gene = ifn6, module = net$colors[ifn6], row.names = NULL)

gene,module
<chr>,<chr>
IFI27,lightgreen
IFI44L,lightgreen
IFIT1,lightgreen
ISG15,turquoise
RSAD2,lightgreen
SIGLEC1,lightgreen


**Explanation**
- As in notebook 05: the module of each of the six interferon-score genes.

**Result.** Four modules read clearly by their hub genes: lightgreen (IFI44L, RSAD2, IFIT1, HERC5, HERC6,
OAS3, IFIT3, OAS2, OAS1: interferon), darkturquoise (CEACAM8, DEFA4, MPO, BPI, CTSG, LTF, MMP8: neutrophil
granule), cyan (CLEC17A, TNFRSF13C, BLNK: B cell) and purple (ORC1, GMNN, RRM2, CDC6, MCM4, CDC45, RAD51:
cell cycle). Five of the six interferon-score genes are in lightgreen; ISG15 is in turquoise.

## Save the modules

In [5]:
in_module <- names(net$colors)[net$colors != "grey"]
module_genes <- data.frame(gene = in_module, module = net$colors[in_module],
                           kME = kME[cbind(in_module, paste0("ME", net$colors[in_module]))], row.names = NULL)
write.csv(module_genes, art("GSE232381", "module_genes.csv"), row.names = FALSE)
saveRDS(list(colors = net$colors, MEs = ME, kME = kME, hubs = hubs), art("GSE232381", "modules.rds"))

**Explanation**
- As in notebook 05. `module_genes.csv` lists this study's own modules; they are not the array modules.